frontend/pkg/frontend/frontend.go

```diff
@@ -507,9 +507,18 @@ func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request
 		}
 	}
 
-	operationDoc, err := f.StartOperation(writer, request, doc, operationRequest)
+	operationDoc := database.NewOperationDocument(operationRequest, doc.Key, doc.InternalID)
```

Comment:
Side question: Any objection or issue if I make a PR to change `Key` -> `ResourceId` in the `ResourceDocument` Type just for readability?